### PR TITLE
Fix PE cleaning

### PR DESCRIPTION
**Pull request by Sugar-pack**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: test/e2e/actions/serverlessprivateendpoint/spe.go [hunk 1]
```diff
@@ -79,12 +83,14 @@ func DeleteSPE(speStatuses []status.ServerlessPrivateEndpoint, providerName prov
 			}
 		}
 	case provider.ProviderAzure:
-		sessionAzure, err := azure.SessionAzure(os.Getenv(networkpeer.SubscriptionID), "spe-test")
+		authorizer, err := auth.NewAuthorizerFromEnvironment()
 		if err != nil {
-			return err
+			return fmt.Errorf("error creating authorizer: %v", err)
```
Comment (igor-karpukhin) [inline test/e2e/actions/serverlessprivateendpoint/spe.go:88]:
```suggestion
			return fmt.Errorf("error creating authorizer: %w", err)
```

**Review by igor-karpukhin — APPROVED** [review]
Looks good 👍 One comment regarding error format

File: .github/workflows/cleanup-all.yml [hunk 2]
```diff
@@ -45,7 +45,7 @@ jobs:
           go-version: '1.19'
 
       - name: Run cleanup VPC
-        if: always() && github.event.inputs.cleanVPC == 'true'
+        if: github.event_name == 'schedule' && github.event.inputs.cleanVPC == 'true'
```
Comment (fabritsius) [inline .github/workflows/cleanup-all.yml:48]: Does this supposed to be `AND`?
```suggestion
        if: github.event_name == 'schedule' || github.event.inputs.cleanVPC == 'true'
```

**Review by fabritsius — APPROVED** [review]
Looks Great 👍 

Just have one question :)

**Review by helderjs — APPROVED** [review]

File: .github/workflows/cleanup-all.yml (hunk 2 above)
Comment (Sugar-pack) [reply .github/workflows/cleanup-all.yml:48]: No

Comment (fabritsius) [reply .github/workflows/cleanup-all.yml]: Yeah, that's what I thought :)

**Pull request merged**